Can't use `rmw_cyclonedds_cpp` with ros2 `humble` because path to `*.so` can't be resolved

Trying to list topics for example, or use any `ros2` command:
```
# ivan @ nixos in /projects/work/nix-ros [12:53:18] 
$ nix develop   
You use 'humble' with 'rmw_cyclonedds_cpp' on domain '1'.

[ivan@nixos:/projects/work/nix-ros]$ ros2 topic list
[ERROR] [1713873209.974130528] [rcl]: Error getting RMW implementation identifier / RMW implementation not installed (expected identifier of 'rmw_cyclonedds_cpp'), with error message 'failed to load shared library 'librmw_cyclonedds_cpp.so' due to dlopen error: librmw_cyclonedds_cpp.so: cannot open shared object file: No such file or directory, at /build/rcutils-release-release-humble-rcutils-5.1.5-1/src/shared_library.c:99, at /build/rmw_implementation-release-release-humble-rmw_implementation-2.8.2-1/src/functions.cpp:65', exiting with 1., at /build/rcl-release-release-humble-rcl-5.3.7-1/src/rcl/rmw_implementation_identifier_check.c:139
```

Here is my `flake.nix` which I use:
```nix
{
  inputs = {
    nix-ros-overlay.url = "github:lopsided98/nix-ros-overlay";
    nixpkgs.follows = "nix-ros-overlay/nixpkgs";  # IMPORTANT!!!
  };
  outputs = { self, nix-ros-overlay, nixpkgs }:
    nix-ros-overlay.inputs.flake-utils.lib.eachDefaultSystem (system:
      let
        ROS_DISTRO = "humble";
        pkgs = import nixpkgs {
          inherit system;
          overlays = [ nix-ros-overlay.overlays.default ];
        };
      in {
        devShells.default = pkgs.mkShell {
          name = "Example project";
          packages = with pkgs.rosPackages.${ROS_DISTRO}; [
            pkgs.colcon
            ros-core
            rmw-cyclonedds-cpp
          ];
          
          RMW_IMPLEMENTATION = "rmw_cyclonedds_cpp";
          ROS_DOMAIN_ID = 1;
          shellHook = ''
          echo "You using '$ROS_DISTRO' with '$RMW_IMPLEMENTATION' on domain '$ROS_DOMAIN_ID'."
          '';
        };
      });

  nixConfig = {
    trusted-substituters = [ "https://ros.cachix.org" ];
    extra-trusted-public-keys = [ "ros.cachix.org-1:dSyZxI8geDCJrwgvCOHDoAfOm5sV1wCPjBkKL+38Rvo=" ];
  };
}
```

I found the `librmw_cyclonedds_cpp.so` exists in the Nix store, but it seems it can't be linked, mb something wrong with `nix-ld`.
```
# ivan @ nixos in /projects/work/nix-ros [13:00:37] C:1
$ locate librmw_cyclonedds_cpp.so
/nix/store/nvgv329v69kmb2j5bvwzmir9xyjy9kyd-ros-humble-rmw-cyclonedds-cpp-1.3.4-r1/lib/librmw_cyclonedds_cpp.so
```

--- a/README.md
+++ b/README.md
@@ -55,10 +55,15 @@ Using the overlay in your `flake.nix`-based project could look like this:
       in {
         devShells.default = pkgs.mkShell {
           name = "Example project";
-          packages = with pkgs.rosPackages.humble; [
+          packages = [
             pkgs.colcon
-            ros-core
-            # ...
+            # ... other non-ROS packages
+            (with pkgs.rosPackages.humble; buildEnv {
+                paths = [
+                    ros-core
+                    # ... other ROS packages
+                ];
+            })
           ];
         };
       });

